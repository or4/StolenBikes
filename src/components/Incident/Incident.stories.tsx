import * as React from 'react';
import { MemoryRouter } from 'react-router';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, number } from '@storybook/addon-knobs';

import { Incident } from './Incident';

import 'reset.css';
import 'App.css';

const stories = storiesOf('Incident', module);
stories.addDecorator(withKnobs);
stories.addDecorator(story => <MemoryRouter initialEntries={['/']}>{story()}</MemoryRouter>);
stories.add('Incident', () => {
    return (
        <Incident
            id={1}
            title={getTitleKnobs()}
            description={getDescriptionKnobs()}
            occurredAt={getAtKnobs()}
            address={getAddressKnobs()}
            imageUrlThumb={getSrcKnobs()}
        />
    );
});

function getTitleKnobs() {
    return text('Header', 'Stolen 2018 VAUN VELO(black)');
}

function getDescriptionKnobs() {
    return text('Description', 'Stolen from my courtyard');
}

function getAtKnobs() {
    return number('OccurredAt', 1532152800);
}

function getAddressKnobs() {
    return text('Address', 'Berlin, 12047, DE');
}

function getSrcKnobs() {
    return text(
        'Source',
        'https://avatars.mds.yandex.net/get-turbo/1505928/2a0000016836daee3ee7d60c44cca211b89b/max_g480_c12_r3x4_pd20',
    );
}
