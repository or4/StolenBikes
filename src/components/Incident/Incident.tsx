import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { IncidentTextPanel, IncidentImage } from 'components';
import { media } from 'ui/media';

interface IProps {
    id: number;
    title?: string;
    description?: string | null;
    address?: string;
    occurredAt?: number;
    imageUrlThumb?: string | null;
}

const Container = styled(Link)`
    display: flex;
    cursor: pointer;
    padding-top: 10px;
    padding-bottom: 10px;
    font-size: 14px;
    outline: none;

    ${media.mobile`
		flex-direction: column;
		border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  	`}

    ${media.tabletNDesktop`
    	min-height: 100px;
	`}

    &:hover {
        background: rgba(0, 0, 0, 0.03);
    }
`;

export const Incident: React.FC<IProps> = (props: IProps) => (
    <Container to={`/case/${props.id}`}>
        <IncidentImage imageUrl={props.imageUrlThumb} />
        <IncidentTextPanel
            title={props.title}
            description={props.description}
            address={props.address}
            occurredAt={props.occurredAt}
        />
    </Container>
);
