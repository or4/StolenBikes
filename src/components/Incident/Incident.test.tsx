import React from 'react';
import { shallow } from 'enzyme';

import { Incident } from './Incident';

describe('Incident', () => {
    it('should render correctly', () => {
        const component = (
            <Incident
                id={1}
                title={'Stolen 2018 VAUN VELO(black)'}
                description={'Stolen from my courtyard'}
                occurredAt={1532152800}
                address={'Berlin, 12047, DE'}
                imageUrlThumb={
                    'https://avatars.mds.yandex.net/get-turbo/1505928/2a0000016836daee3ee7d60c44cca211b89b/max_g480_c12_r3x4_pd20'
                }
            />
        );

        const container = shallow(component);
        expect(container).toMatchSnapshot();
        container.unmount();
    });

    it('without some fields', () => {
        const component = <Incident id={1} />;

        const container = shallow(component);
        expect(container).toMatchSnapshot();
        container.unmount();
    });
});
