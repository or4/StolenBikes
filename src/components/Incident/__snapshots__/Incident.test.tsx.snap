// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incident should render correctly 1`] = `
<Styled(Link)
  to="/case/1"
>
  <IncidentImage
    imageUrl="https://avatars.mds.yandex.net/get-turbo/1505928/2a0000016836daee3ee7d60c44cca211b89b/max_g480_c12_r3x4_pd20"
  />
  <IncidentTextPanel
    address="Berlin, 12047, DE"
    description="Stolen from my courtyard"
    occurredAt={1532152800}
    title="Stolen 2018 VAUN VELO(black)"
  />
</Styled(Link)>
`;

exports[`Incident without some fields 1`] = `
<Styled(Link)
  to="/case/1"
>
  <IncidentImage />
  <IncidentTextPanel />
</Styled(Link)>
`;
